src/Shared/HttpSys/RequestProcessing/HeaderEncoding.cs

```diff
@@ -27,4 +28,21 @@ internal static byte[] GetBytes(string myString)
     {
         return Encoding.GetBytes(myString);
     }
+
+    internal static Span<byte> GetBytes(string myString, IBufferWriter<byte> writer)
+    {
+        // Compute the maximum amount of bytes needed for the given string.
+        // Include an extra byte for the null terminator.
+        Span<byte> buffer = writer.GetSpan(Encoding.GetMaxByteCount(myString.Length) + 1);
```

Comment:
This (and everything else in this file) should use Encoding.UTF8 directly, and the static encoding cache should be deleted.

Using Encoding.UTF8 directly allows better devirtualization.

Reply:
The static allocation above sets `encoderShouldEmitUTF8Identifier: false`, whereas `Encoding.UTF8` sets it to `encoderShouldEmitUTF8Identifier: true`. I can however change the type to be `UTF8Encoding` instead.

Reply:
`encoderShouldEmitUTF8Identifier` matters only when the encoding is used for streaming (ie you get encoder and feed a stream of data into the encoder). None of the encoding uses in this type are streaming. `Encoding.UTF8` will work just fine here. 

> I can however change the type to be UTF8Encoding instead.

This helps a little, but it still worse than using `Encoding.UTF8` directly.